src/search.cpp

```diff
@@ -278,6 +285,8 @@ int Search::getEstimatedMatches() const
       return mset.get_matches_estimated();
     } catch(Xapian::QueryParserError& e) {
       return 0;
+    } catch(Xapian::DatabaseError& e) {
```

Comment:
Shouldn't this be `Xapian::DatabaseCorruptError`? Though the latter seems to be the only subclass of `Xapian::DatabaseError` that we may expect to catch here (and in most other locations changed by this PR), if another type of a `Xapian::DatabaseError` is introduced in the future it may be wrongly converted to `zim::ZimFileFormatError`. The question then may seem to be what should we do with such a new type of a Xapian exception, but it can be answered right away since there already exist other types of Xapian exceptions not derived from `Xapian::DatabaseError`. If we let such exceptions propagate outside `libzim`, doing the same for a future new type of error shouldn't be a problem. However, under such logic why do we have to make an exception (pun intended) for `DatabaseCorruptError` exceptions and convert them to `zim::ZimFileFormatError`?

Reply:
I think we should catch at least `Xapian::DatabaseCorruptError` and `Xapian::DatabaseVersionError`. The common parent exception is `Xapian::DatabaseError`.

It is a trade of between simplification of code (and maybe catch too many things) and specific catch (but doubled `catch` block)